### ISSUE TITLE
Error when running Tensor::backward() with all dense/sparse tensors on CPU device

### ISSUE THREAD

**Issue by harvey298** [LaurentMazare/tch-rs#701]
When running `Tensor::backward()` with all dense/sparse tensors on CPU device, the following error occurs:

```
Torch("Could not run 'aten::view' with arguments from the 'SparseCPU' backend. This could be because the operator doesn't exist for 
this backend, or was omitted during the selective/custom build process (if using custom build). If you are a Facebook employee using PyTorch on mobile, please visit https://fburl.com/ptmfixes for possible resolutions. 'aten::view' is only available for these backends: [CPU, Meta, QuantizedCPU, MkldnnCPU, NestedTensorCPU, BackendSelect, Python, FuncTorchDynamicLayerBackMode, Functionalize, Named, Conjugate, Negative, ZeroTensor, 
ADInplaceOrView, AutogradOther, AutogradCPU, AutogradCUDA, AutogradHIP, AutogradXLA, AutogradMPS, AutogradIPU, AutogradXPU, AutogradHPU, AutogradVE, AutogradLazy, AutogradMeta, AutogradMTIA, AutogradPrivateUse1, AutogradPrivateUse2, AutogradPrivateUse3, AutogradNestedTensor, Tracer, AutocastCPU, AutocastCUDA, FuncTorchBatched, FuncTorchVmapMode, Batched, VmapMode, FuncTorchGradWrapper, PythonTLSSnapshot, FuncTorchDynamicLayerFrontMode, PythonDispatcher]")
```

All tensors used have `requires_grad` set to true.

It seems that the SparseCPU backend does not support the view operator required for the backward pass. Is there any way to change the backend to avoid this error?

**Comment by harvey298** [comment]
Fixed, my solution was to use the `to_dense_backward`

like this: `let loss = loss.to_dense_backward(&loss);`
